src/nvim/ex_docmd.c

```diff
@@ -4906,18 +4906,13 @@ static void ex_restart(exarg_T *eap)
     set_vim_var_list(VV_ARGV, argv_cpy);
   }
 
-  bool confirm = (p_confirm || (cmdmod.cmod_flags & CMOD_CONFIRM));
-  if (confirm && check_changed_any(false, false)) {
-    return;
-  }
+  char *quit_cmd = (eap->do_ecmd_cmd) ? eap->do_ecmd_cmd : "qall";
 
-  char *quit_cmd;
-  if (eap->do_ecmd_cmd) {
-    quit_cmd = eap->do_ecmd_cmd;
-  } else if (confirm) {
-    quit_cmd = "qall";
-  } else {
-    quit_cmd = "qall!";
+  // Prepend "confirm " to cmd if :confirm is used
+  if (cmdmod.cmod_flags & CMOD_CONFIRM) {
+    char *quit_cmd_copy = NULL;
+    quit_cmd_copy = concat_str("confirm ", quit_cmd);
```

Comment:
I'm quite new to C, so I'd like feedback on whether this is the correct way to prepend a static string. I tested it and it works fine, but I'm still not the best at catching memory issues.